### PR TITLE
Reload table form and set not null

### PR DESCRIPTION
**Pull request by zaucker**
- allow setting form values to 0 and `''`
- reload TableForm on data change (e.g., filters)

### CODE REVIEW

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Table.js [hunk 1]
```diff
@@ -203,8 +203,10 @@ qx.Class.define('callbackery.ui.plugin.Table', {
                 model.reloadData();
             });
             this._form.addListener('changeData',function(e){
+                console.log('_form=', this._form);
```
Comment (oetiker) [inline lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Table.js:206]:
```suggestion
```

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Table.js [hunk 2]
```diff
@@ -205,6 +205,7 @@ qx.Class.define('callbackery.ui.plugin.Table', {
             this._form.addListener('changeData',function(e){
                 model.setFormData(e.getData());
                 if (this._loading == 0){ // only reload when data has been changed by human
+                    this._loadDataReadOnly();
```
Comment (oetiker) [inline lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Table.js:208]: hmmm where the table is a descendant of the form class, and thus has a _loadDataReadOnly method, I would still want to call the _loadDataReadOnly method of the associated form, as it is the form I want to reload and not the table.
```suggestion
                    this._form._loadDataReadOnly();
```

Comment (oetiker) [reply lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Table.js:208]: also a few lines below, there is another reloadData, one issued in response to clicking a reload button. I guess there a this._form__loadDataReadOnly would also be warrented.

**Review by oetiker — CHANGES_REQUESTED** [review]
as described

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Table.js (hunk 2 above)
Comment (zaucker) [reply lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Table.js:208]: `this._form` is an instance of `callbackery.ui.form.Auto` which does not have the `_loadDataReadOnly` method. 

`_loadDataReadOnly` is only defined in `callbackery.ui.plugin.Form` (the parent of `callbackery.ui.plugin.Table`). As far as I can tell `Table` extends `Form` and does not include a separate instance of `Form` (compose pattern).

Comment (oetiker) [reply lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Table.js:208]: ah :see_no_evil: for the _form ...

the comment on the missing second invocation of _loadDataReadonly remains ...

Comment (zaucker) [reply lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Table.js:208]: Just pushed :-)

**Review by oetiker — APPROVED** [review]

**Pull request merged**